eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -1,61 +1,32 @@
-parameters:
-  # Use the variable name itself (no $() or other wrapping syntax). This is
-  # because we use runtime and macro expressions for conditions and script
-  # invocations
-  DependenciesVariableName: vcpkg.deps
-
 steps:
   - pwsh: |
       $TargetPath = "$(Agent.TempDirectory)/vcpkg"
       Remove-Item -Path $TargetPath -Recurse -Force -ErrorAction Ignore
       New-Item -ItemType Directory -Path $TargetPath  -Force
 
       $VcpkgCommit = $(Get-Content eng/vcpkg-commit.txt)
+      $VcpkgCacheMode = If (Test-Path 'env:VCPKG_CACHE_SAS') {"readwrite"} Else {"read"}
```

Comment:
Without the if I suggested in the other comment this will always exist as a variable and thus this Test-Path will always return true.

Reply:
ah, I made a test locally and Test-Path returns false if the environment variable is set with empty string.  But yes, let me try what you suggest.